### PR TITLE
[cli] Fix `vc link` not respecting `--yes`

### PR DESCRIPTION
**Pull request by MatthewStanciu**
As of 28.0.0, `vc link` stopped respecting `--yes` and would wait for prompt if it found a Git repository to connect. This PR passes `autoConfirm` through to the Git prompt, so that `vc link` continues to respect `--yes`.

Here's what `vc link` looks like after this PR if a user has a local Git repository:

Screen Shot 2022-08-16 at 10 05 56 AM

Not sure if we want to remove the log messages to be consistent with the rest of `vc link`?

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

**Review by cb1kenobi — APPROVED** [review]
LGTM

File: packages/cli/src/util/link/add-git-connection.ts [hunk 1]
```diff
@@ -81,13 +85,17 @@ async function addSingleGitRemote(
...
+  if (!shouldConnect) {
+    shouldConnect = await promptGitConnectSingleUrl(
+      client,
+      project,
+      remoteName,
+      remoteUrl,
+      replace
+    );
+  }
   return handleOptions(shouldConnect, client, org, project, settings, repoInfo);
```
Comment (EndangeredMassa) [inline packages/cli/src/util/link/add-git-connection.ts]: **suggestion (non-blocking):** rearrange logic to make it clearer

```suggestion
  let shouldConnectOption: string | undefined;
  if (autoConfirm) {
    shouldConnectOption = 'yes';
  } else {
    shouldConnectOption = await promptGitConnectSingleUrl(
      client,
      project,
      remoteName,
      remoteUrl,
      replace
    );
  }
  return handleOptions(shouldConnectOption, client, org, project, settings, repoInfo);
```

I had to read this several times to understand what was going on. The `option` parameter of `promptGitConnectSingleUrl` made it a bit more confusing for me.

I suggest rearranging this logic to clarify.

File: packages/cli/src/util/link/add-git-connection.ts [hunk 2]
```diff
@@ -96,12 +104,13 @@ async function addMultipleGitRemotes(
...
-  settings: ProjectSettings
+  settings: ProjectSettings,
+  autoConfirm: Boolean
 ) {
-  client.output.log('Found multiple Git remote URLs in Git config.');
-  const remoteUrlOrOptions = await promptGitConnectMultipleUrls(
-    client,
-    remoteUrls
-  );
+  let remoteUrlOrOptions = autoConfirm ? Object.values(remoteUrls)[0] : '';
```
Comment (EndangeredMassa) [inline packages/cli/src/util/link/add-git-connection.ts]: **suggestion (non-blocking):** use more elaborate defaults

I suggest defaulting like so:

- if only 1 remote
  - use it
- else (many remotes)
  - if `origin` exists
    - use `origin` 
  - else
    - use first remote

Changing this later should probably be considered a breaking change. We may want to address this now.

File: packages/cli/src/util/link/add-git-connection.ts [hunk 3]
```diff
@@ -96,12 +104,13 @@ async function addMultipleGitRemotes(
...
+  if (!remoteUrlOrOptions) {
+    client.output.log('Found multiple Git remote URLs in Git config.');
+    remoteUrlOrOptions = await promptGitConnectMultipleUrls(client, remoteUrls);
+  }
   return handleOptions(remoteUrlOrOptions, client, org, project, settings);
```
Comment (EndangeredMassa) [inline packages/cli/src/util/link/add-git-connection.ts]: **suggestion (non-blocking):** rearrange logic for clarity

The use of `remoteUrlOrOptions` for multiple values is confusing.

```suggestion
  let remoteUrl: string | undefined;
  if (autoConfirm) {
    remoteUrl = Object.values(remoteUrls)[0];
  } else {
    client.output.log('Found multiple Git remote URLs in Git config.');
    remoteUrl = await promptGitConnectMultipleUrls(client, remoteUrls);
  }
  return handleOptions(remoteUrl, client, org, project, settings);
```

**Review by EndangeredMassa — APPROVED** [review]

File: packages/cli/src/util/link/add-git-connection.ts (hunk 2 above)
Comment (MatthewStanciu) [reply packages/cli/src/util/link/add-git-connection.ts]: Thanks 👍 I think I addressed this in my latest commit, but let me know if you had something else in mind.

File: packages/cli/test/unit/commands/link.test.ts [hunk 4]
```diff
@@ -306,5 +306,28 @@ describe('link', () => {
...
+        client.setArgv('--yes');
+        const linkPromise = link(client);
+        expect(client.stderr).not.toOutput('Do you want to connect "origin"');
+        await expect(client.stderr).toOutput('Linked to');
+        await expect(linkPromise).resolves.toEqual(0);
+      } finally {
+        await fs.rename(join(cwd, '.git'), join(cwd, 'git'));
+        process.chdir(originalCwd);
+      }
+    });
```
Comment (EndangeredMassa) [inline packages/cli/test/unit/commands/link.test.ts:331]: **suggestion (soft-blocking):** add test for multiple remotes

In this or a follow-up PR: Can we add a test for multiple remotes that prefer `origin` when `origin` is not the first item in the list?

**Review by EndangeredMassa — APPROVED** [review]

File: packages/cli/test/unit/commands/link.test.ts (hunk 4 above)
Comment (MatthewStanciu) [reply packages/cli/test/unit/commands/link.test.ts:331]: Added test!

**Review by EndangeredMassa — APPROVED** [review]

**Pull request merged**